src/cryptonote_basic/miner.cpp

```diff
@@ -846,40 +846,81 @@ namespace cryptonote
 
     #elif defined(__linux__)
 
-      // i've only tested on UBUNTU, these paths might be different on other systems
-      // need to figure out a way to make this more flexible
-      std::string power_supply_path = "";
-      const std::string POWER_SUPPLY_STATUS_PATHS[] = 
-      {
-        "/sys/class/power_supply/ACAD/online",
-        "/sys/class/power_supply/AC/online",
-        "/sys/class/power_supply/AC0/online",
-        "/sys/class/power_supply/ADP0/online"
-      };
+      // Use the power_supply class http://lxr.linux.no/#linux+v4.10.1/Documentation/power/power_supply_class.txt
+      std::string power_supply_class_path = "/sys/class/power_supply";
 
-      for(const std::string& path : POWER_SUPPLY_STATUS_PATHS)
+      boost::tribool is_plugged = boost::logic::tribool(boost::logic::indeterminate);
```

Comment:
I think a lot of the comments from @jtgrassie start here. The function is called `on_battery_power`, and the status variable for the loop is named `is_plugged`. These are opposing concepts. I think the logic would be easier to understand if they matched.